### PR TITLE
✨ geo service enhancement

### PR DESCRIPTION
**Pull request by jpettitt**
Adds functionality to `geo` service

1. `matchedCountryGroups` array replaces `ISOCountryGroups`.  The old name is deprecated but retained to avoid version skew and will be removed at a later date when consuming extensions are updated.
1. add `allISOCountryGroups` array to allow extension to detect if they are configured with an impossible group.
1. adds `isInCountryGroup()` api.  Accepts a country group or comma separated list of country groups and returns an enum `IN`, `NOT_IN` or `NOT_DEFINED`.  This removes the need for extensions to include their own group testing logic.

### CODE REVIEW

File: extensions/amp-geo/0.1/amp-geo.js [hunk 1]
```diff
@@ -46,6 +46,14 @@ import {parseJson} from '../../../src/json';
 import {user} from '../../../src/log';
 import {waitForBodyPromise} from '../../../src/dom';
 
+/**
+ * @enum {number}
+ */
+export const GEO_IN_GROUP = {
+  NOT_DEFINED: 0,
```
Comment (zhouyx) [inline extensions/amp-geo/0.1/amp-geo.js]: I learned my lesson last time to not use `0` as a enum value. the check for `if (!value)` is error prone, can we start with 1 instead : )

File: extensions/amp-geo/0.1/amp-geo.js [hunk 2]
```diff
@@ -268,9 +280,50 @@ export class AmpGeo extends AMP.BaseElement {
           break;
       }
 
-      return {ISOCountry: self.country_, ISOCountryGroups: self.matchedGroups_};
+      return {
+        ISOCountry: self.country_,
+        matchedISOCountryGroups: self.matchedGroups_,
+        allISOCountryGroups: Object.keys(config.ISOCountryGroups || {}),
```
Comment (zhouyx) [inline extensions/amp-geo/0.1/amp-geo.js]: nit: use `this.definedGroups_` instead?

File: extensions/amp-geo/0.1/amp-geo.js (hunk 1 above)
Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: done

File: extensions/amp-geo/0.1/amp-geo.js (hunk 2 above)
Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: done

**Pull request merged**